jscatter/_cli.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import os
+import shutil
+import sys
+from pathlib import Path
+
+from jscatter import __version__
+
+_DEV = False
+
+
+def check_uv_available():
+    if shutil.which("uv") is None:
+        print("Error: 'uv' command not found.", file=sys.stderr)
+        print("Please install 'uv' to run `jscatter demo` entrypoint.", file=sys.stderr)
+        print(
+            "For more information, visit: https://github.com/astral-sh/uv",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+
+def run_notebook(notebook_path: Path):
+    check_uv_available()
+
+    command = [
+        "uvx",
+        "--python",
+        "3.12",
+        "--from",
+        "jupyter-core",
+        "--with",
+        "jupyterlab",
+        "--with",
+        "." if _DEV else f"jupyter-scatter=={__version__}",
+        "jupyter",
+        "lab",
+        str(notebook_path),
+    ]
+
+    try:
+        os.execvp(command[0], command)
+    except OSError as e:
+        print(f"Error executing {command[0]}: {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="jupyter-scatter")
+    subparsers = parser.add_subparsers(dest="command", help="Available commands")
+    subparsers.add_parser("demo", help=f"Run the demo notebook in JupyterLab")
+    args = parser.parse_args()
+
+    notebook_path = Path("notebooks/demo.ipynb")
```

Comment:
I think you'll want to make this path resolved relative to this file:

```suggestion
    notebook_path = Path(__file__) "../notebooks/demo.ipynb"
```

and also make sure that the notebook is included in the hatchling wheel and sdist.


My guess is that right now if you run:

```sh
uv build
uvx --from=./dist/jupyter-scatter-<wheel> jupyter-scatter demo
```

where will be an error.

Reply:
Thanks for the catch!

Reply:
Turns out that including the file in the wheel is more complicated than expected. I'm not a Python packaging expert but even if a folder outside the actual package (`jscatter`) is included in the wheel, it ends up not being accessible after the wheel is installed. I ended up dynamically downloading the notebook from GitHub to get around the packaging issue.